include/flux/op/cartesian_product.hpp

```diff
@@ -242,6 +284,16 @@ struct sequence_traits<detail::cartesian_product_adaptor<Bases...>>
     {
         return read_(flux::move_at_unchecked, self, cur);
     }
+
+    template <typename Self, typename Function>
+    static constexpr auto for_each_while(Self& self, Function&& func)
+        -> cursor_t<Self>
+    {
+        bool keep_going = true;
+        cursor_t<Self> cur;
```

Comment:
Now whenever anybody asks me why cursors for multipass sequences need to be default-constructible, I'll have a good answer :)

Reply:
Ah, I just assumed they were!